src/main/java/sirius/web/http/session/UserAgent.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.web.http.session;
+
+import sirius.kernel.commons.Strings;
+
+import java.util.regex.Pattern;
+
+/**
+ * Provides information about user agent and used device for given user agent.
+ * <p>
+ * This class only provides valid information for requests by browsers or clients which do transmit an user agent. If
+ * no user agent is transmitted, the client is considered as desktop.
+ * <p>
+ * Assumptions of device properties based on user agents might not be accurate all the time as some browsers transmit
+ * some special user agents and user agents for different browsers and devices might change over time.
+ */
+public class UserAgent {
+
+    private static final String IPHONE = "iphone";
+    private static final String IPAD = "ipad";
+    private static final String IPOD = "ipod";
+    private static final Pattern ANDROID_PHONE_PATTERN = Pattern.compile("(?=.*\\bandroid\\b)(?=.*\\bmobile\\b)");
+    private static final String ANDROID_TABLET = "android";
+    private String userAgentString;
+
+    private boolean userAgentParsed = false;
+
+    private boolean phone = false;
+    private boolean tablet = false;
+    private boolean android = false;
+    private boolean iOS = false;
+
+    public UserAgent(String userAgentString) {
+        this.userAgentString = userAgentString;
+    }
+
+    protected void parseUserAgent() {
+        userAgentParsed = true;
+        if (Strings.isEmpty(userAgentString)) {
```

Comment:
why not call parseUserAgent in the constructor....it will be called eventually....